pkg/cloudprovider/aws/instancetypes.go

```diff
@@ -146,22 +145,25 @@ func (p *InstanceTypeProvider) getAllInstanceTypes(ctx context.Context) ([]*ec2.
 }
 
 // filterFrom returns a filtered list of instance types based on the provided resource constraints
-func (p *InstanceTypeProvider) filterFrom(instanceTypes []*packing.Instance, constraints *cloudprovider.Constraints, zones []string) []*packing.Instance {
+func (p *InstanceTypeProvider) filterFrom(instanceTypes []*packing.Instance, constraints *AWSConstraints, zones []string) []*packing.Instance {
 	filtered := []*packing.Instance{}
-	architecture := utils.NormalizeArchitecture(constraints.Architecture)
-
 	for _, instanceTypeInfo := range instanceTypes {
-		if (len(constraints.InstanceTypes) == 0 || functional.ContainsString(constraints.InstanceTypes, *instanceTypeInfo.InstanceType)) &&
-			(len(constraints.InstanceTypes) != 0 || p.isDefaultInstanceType(instanceTypeInfo)) &&
-			(architecture == nil || functional.ContainsString(aws.StringValueSlice(instanceTypeInfo.ProcessorInfo.SupportedArchitectures), *architecture)) &&
-			functional.ContainsString(aws.StringValueSlice(instanceTypeInfo.SupportedUsageClasses), "on-demand") &&
-			(len(zones) == 0 || len(functional.IntersectStringSlice(instanceTypeInfo.Zones, zones)) > 0) {
+		if constraints.Constraints == nil ||
```

Comment:
Can we just create an invariant where constraints is never nil?

Reply:
fixed